pkg/projectfile/projectfile.go

```diff
@@ -468,9 +468,16 @@ func (p *Project) Init() error {
 	if p.parsedURL.LegacyCommitID != "" {
 		// Migrate from commitID in activestate.yaml to .activestate/commit file.
 		// Writing to disk during Parse() feels wrong though.
-		if err := localcommit.Set(filepath.Dir(p.Path()), p.parsedURL.LegacyCommitID); err != nil {
+		projectDir := filepath.Dir(p.Path())
+		if err := localcommit.Set(projectDir, p.parsedURL.LegacyCommitID); err != nil {
 			return errs.Wrap(err, "Could not create local commit file")
 		}
+		if fileutils.DirExists(filepath.Join(projectDir, ".git")) {
+			err := localcommit.AddToGitIgnore(projectDir)
+			if err != nil {
+				multilog.Error("Unable to add local commit file to .gitignore: %v", err)
```

Comment:
Currently, project parsing has no outputer for notices, so we cannot notify the user to manually add this file like we do with `checkout` and `init`. I've updated the ACs of DX-2263 to support this.